Sources/Build/BuildOperation.swift

```diff
@@ -461,11 +461,32 @@ public final class BuildOperation: PackageStructureDelegate, SPMBuildCore.BuildS
         let buildResultBuildPlan = buildOutputs.contains(.buildPlan) ? try buildPlan : nil
         let buildResultReplArgs = buildOutputs.contains(.replArguments) ? try buildPlan.createREPLArguments() : nil
 
+        let artifacts: [(String, PluginInvocationBuildResult.BuiltArtifact)]?
+        if buildOutputs.contains(.builtArtifacts) {
+            let builtProducts = try buildPlan.buildProducts
+            artifacts = try builtProducts.compactMap {
+                switch $0.product.type {
+                case .library(let kind):
+                    return try ($0.product.name, .init(
+                        path: $0.binaryPath.pathString,
+                        kind: (kind == .dynamic) ? .dynamicLibrary : .staticLibrary)
```

Comment:
Suggest using a switch instead of a ternary so we don't silently default new cases to staticLibrary in the future

Reply:
fixed, I copied this around and didn't look it over carefully